### PR TITLE
Add annotation for content encoding/media type

### PR DESCRIPTION
**Pull request by huber-nicolas**

### CODE REVIEW

**Comment by huber-nicolas** [comment]
I think the remaining abaplint findings are no valid findings

**Comment by larshp** [comment]
unit test will be fixed with https://github.com/SAP/abap-file-formats-tools/pull/229

File: src/zcl_aff_writer_json_schema.clas.abap [hunk 1]
```diff
@@ -302,6 +315,48 @@ CLASS zcl_aff_writer_json_schema IMPLEMENTATION.
...
+      ELSE.
+        log->add_warning( message_text = zif_aff_log=>co_msg129 component_name = fullname_of_type ).
+      ENDIF.
+    ENDIF.
+
+  ENDMETHOD.
+
+  METHOD write_content_media_type.
+
+    IF abap_doc-content_media_type IS NOT INITIAL.
```
Comment (wurzka) [inline src/zcl_aff_writer_json_schema.clas.abap:336]: Same as above

File: src/zcl_aff_writer_json_schema.clas.testclasses.abap [hunk 2]
```diff
@@ -504,7 +504,12 @@ CLASS ltcl_json_writer_abap_doc DEFINITION FINAL FOR TESTING
       get_extrema_integer1 FOR TESTING RAISING cx_static_check,
       calculate_max_length FOR TESTING RAISING cx_static_check,
       original_description FOR TESTING RAISING cx_static_check,
-      struc_with_own_enum_values FOR TESTING RAISING cx_static_check.
+      struc_with_own_enum_values FOR TESTING RAISING cx_static_check,
+    content_encoded_integer FOR TESTING RAISING cx_static_check,
```
Comment (wurzka) [inline src/zcl_aff_writer_json_schema.clas.testclasses.abap:508]: Indentation

**Review by wurzka — CHANGES_REQUESTED** [review]
Functionally, it looks good to me.

See my comments about the code

**Comment by larshp** [comment]
overall, what is this feature related to? 

I've been arguing to collect the JSON data in few files, but I'd not expect to encode other filetypes inside the JSON as the examples in https://json-schema.org/understanding-json-schema/reference/non_json_data.html#id4

If there is a html or png file, just encode it as another file as per https://github.com/SAP/abap-file-formats/blob/main/docs/specification.md#file-names ?

**Comment by larshp** [comment]
does the unit test pass in a real system?

**Comment by schneidermic0** [comment]
> as in openapi REST? see https://github.com/abap-openapi/

I am not sure what you mean.

File: src/zif_aff_log.intf.abap [hunk 3]
```diff
@@ -19,6 +19,8 @@ INTERFACE zif_aff_log
     co_msg126 TYPE string VALUE `For required fields, a default handling is not possible` ##NO_TEXT,
     co_msg127 TYPE string VALUE `Elements of type enum should be required or have a default` ##NO_TEXT,
     co_msg128 TYPE string VALUE `Components on top level (except format_version) should be structures or tables` ##NO_TEXT,
+    co_msg132 TYPE string VALUE `Only elements of character-like type can have annotation $contentEncoding or $contentMediaType` ##NO_TEXT,
+    co_msg133 TYPE string VALUE `The content encoding is unknown` ##NO_TEXT,
     co_msg129 TYPE string VALUE `Fields of a different type than integer should not use the annototation $multipleOf` ##NO_TEXT.
```
Comment (wurzka) [inline src/zif_aff_log.intf.abap:24]: Maybe we should keep the correct order

**Review by wurzka — APPROVED** [review]
LGTM :)

**Comment by wurzka** [comment]
@larshp Failing unit test is green in our system. We do not know why it is failing here

File: test/abap_transpile.json [hunk 4]
```diff
@@ -21,6 +21,7 @@
     "unknownTypes": "runtimeError",
     "extraSetup": "../test/setup.mjs",
     "skip": [
+      {"object": "ZCL_AFF_ABAP_DOC_PARSER", "class": "ltcl_aff_abap_doc_parser", "method": "content_media_multiple_entries"},
```
Comment (huber-nicolas) [inline test/abap_transpile.json:24]:
```suggestion
      {"object": "ZCL_AFF_ABAP_DOC_PARSER", "class": "ltcl_aff_abap_doc_parser", "method": "content_media_multiple_entries"},
      {"object": "ZCL_AFF_ABAP_DOC_PARSER", "class": "ltcl_aff_abap_doc_parser", "method": "content_encod_multiple_entries"},
```

**Pull request merged**